totp/totp.py

```diff
@@ -9,7 +14,17 @@
 
 def generate_secret() -> str:
     """Returns a random 20-symbol base58 string"""
-    return "abcdeabcdeabcdeabcd" + str(int(time.time_ns()) % 9)
+    # alphabet = (string.ascii_letters + string.digits).translate({ord(c): '' for c in "0OlI"})
+    # secret = ''.join(secrets.choice(alphabet) for i in range(20))
+
+    alphabet = secrets.token_bytes(30)
+    secret = base58.b58encode(alphabet).decode('utf-8')[:20]
```

Comment:
workable, but:
a bit too much random bytes are generated. 20 symbols of base58 is circa 14 bytes of random binary

Reply:
and edge case: `secret` could (very very rarely) be less than 20 bytes long, need to be extended in that case

Reply:
I'd sugest adding something along the lines of `if secret's lenght is <20 pad the secret with something until it's 20'